tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,375 @@
+        deploy_flavor=settings.flavors.default,
+    ) as chost:
+        yield chost
+
+
+@pytest.mark.parametrize(
+    'upgrade_path',
+    [
+        # {'source_version': RHEL7_VER, 'target_version': RHEL8_VER},
+        {'source_version': RHEL8_VER, 'target_version': RHEL9_VER},
+    ],
+    ids=lambda upgrade_path: f'{upgrade_path["source_version"]}'
+    f'_to_{upgrade_path["target_version"]}',
+)
+@pytest.mark.usefixtures(
+    'leapp_sat_content',
+    'register_host_with_satellite',
+    'verify_target_repo_on_satellite',
+    'precondition_check_upgrade_and_install_leapp_tool',
+    'fix_inhibitors',
+)
+def test_leapp_upgrade_rhel(
+    module_target_sat,
+    custom_leapp_host,
+    upgrade_path,
+):
+    """Test to upgrade RHEL host to next major RHEL Realse with Leapp Preupgrade and Leapp Upgrade
+    Job templates
+
+    :id: 8eccc689-3bea-4182-84f3-c121e95d54c3
+
+    :Steps:
+        1. Import a subscription manifest and enable, sync source & target repositories
+        2. Create LCE, Create CV, add repositories to it, publish and promote CV, Create AK, etc.
+        3. Register content host with AK
+        4. Varify target rhel repositories are enable on Satellite
+        5. Update all packages, install leapp tool and fix inhibitors
+        6. Run Leapp Preupgrade and Leapp Upgrade job template
+
+    :expectedresults:
+        1. Update RHEL OS major version to another major version
+
+    """
+    logger.info('Running test for upgrade path %s', upgrade_path)
+    old_ver = custom_leapp_host.os_version.major
+    # 6. Run LEAPP-PREUPGRADE Job Template-
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run preupgrade via Leapp"'})[0]
+        .id
+    )
+    job = module_target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'targeting_type': 'static_query',
+            'search_query': f'name = {custom_leapp_host.hostname}',
+        },
+    )
+    module_target_sat.wait_for_tasks(
+        f'resource_type = JobInvocation and resource_id = {job["id"]}', poll_timeout=1800
+    )
+    result = module_target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
+    # In case of Upgrade Path A - Rehl-7 to Rhel-8
+    # Inhibitors should fix after running LEAPP-PREUPGRADE Job Template
+    if upgrade_path['source_version'].split('.')[0] == '7':
+        logger.info('Fixing inhibitory for upgrade path : %s', upgrade_path)
+        # 1. Leapp detected loaded kernel drivers which have been removed in RHEL 8
+        custom_leapp_host.run('rmmod floppy')
+        custom_leapp_host.run('rmmod pata_acpi')
+        # 2. Missing required answers in the answer file
+        leapp_report_path = "/var/log/leapp/leapp-report.txt"
+        grep_cmd = f"grep 'Title: Missing required answers in the answer file' {leapp_report_path}"
```

Comment:
Use single quotes wherever possible to keep it consistent
```suggestion
        leapp_report_path = '/var/log/leapp/leapp-report.txt'
        grep_cmd = f'grep "Title: Missing required answers in the answer file" {leapp_report_path}'
```